### PR TITLE
feat: make Percentage only positive

### PR DESCRIPTION
**Pull request by Gounlaf**
:wave: 

As discussed https://github.com/dlemstra/Magick.NET/discussions/1568#discussioncomment-8635702, this PR make Percentage positive only (including zero).

Therefore:
- all guards checking if percentage value is negative has been removed
- all tests where `Percentage` was initialized with negative value has been removed

Regards.

### CODE REVIEW

File: src/Magick.NET.Core/Types/Percentage.cs [hunk 1]
```diff
@@ -18,14 +18,22 @@ namespace ImageMagick;
     /// </summary>
     /// <param name="value">The value (0% = 0.0, 100% = 100.0).</param>
     public Percentage(double value)
-        => _value = value;
+    {
+        Throw.IfNegative(nameof(value), value);
```
Comment (Gounlaf) [inline src/Magick.NET.Core/Types/Percentage.cs:22]: Maybe I should have used `Throw.IfOutOfRange` with [0.0,100.0] range?

Comment (Gounlaf) [reply src/Magick.NET.Core/Types/Percentage.cs:22]: I may have found a counter argument to do it:

https://github.com/dlemstra/Magick.NET/blob/483e8753f16f5b2401c9eb447e5e9e091fdb8749/tests/Magick.NET.Tests/MagickImageTests/TheResizeMethod.cs#L94-L105

Comment (Gounlaf) [reply src/Magick.NET.Core/Types/Percentage.cs:22]: Same for `Scale` method.

Comment (dlemstra) [reply src/Magick.NET.Core/Types/Percentage.cs:22]: Good that I have some unit tests for that :smile:. Percentages above 100% are allowed.

Comment (Gounlaf) [reply src/Magick.NET.Core/Types/Percentage.cs:22]: At least description of Percentage should be updated 😅

Comment (dlemstra) [reply src/Magick.NET.Core/Types/Percentage.cs:22]: Maybe we should change it to `(0% = 0.0, 100% = 100.0, 142.42% = 142.42)`?

**Pull request merged**